core/client/app/controllers/settings/apps.js

```diff
@@ -0,0 +1,91 @@
+import Ember from 'ember';
+import DS from 'ember-data';
+import SettingsSaveMixin from 'ghost/mixins/settings-save';
+import ValidationEngine from 'ghost/mixins/validation-engine';
+
+const {
+    Controller,
+    computed,
+    inject: {controller},
+    isBlank
+} = Ember;
+const {Errors} = DS;
+const emberA = Ember.A;
+
+export const Slack = Ember.Object.extend(ValidationEngine, {
+    url: '',
+    channel: '',
+    name: '',
+    icon: 'ghost',
+    isActive: false,
+
+    validationType: 'settings',
+
+    isComplete: computed('channel', 'url', function () {
+        let {channel, url} = this.getProperties('channel', 'url');
+
+        return !isBlank(channel) && !isBlank(url);
+    }),
+
+    isBlank: computed('channel', 'url', function () {
+        let {channel, url} = this.getProperties('channel', 'url');
+
+        return isBlank(channel) && isBlank(url);
+    }),
+
+    init() {
+        this._super(...arguments);
+        this.set('errors', Errors.create());
+        this.set('hasValidated', emberA());
+    }
+});
+export default Controller.extend(SettingsSaveMixin, {
+    slackController: controller('settings.apps.slack'),
+    newSlackConfig: null,
+
+    slack: computed('model.slack', function () {
+        let slackItem;
+
+        try {
+            [ slackItem ] = JSON.parse(this.get('model.slack') || [{}]);
```

Comment:
Either that or the `|| [{}]` can be removed entirely as any parse errors would be caught by the catch statement.